src/react/stats.js

```diff
@@ -11,10 +11,18 @@ export const stats = {
   state: createState(),
 
   set(replacedDep) {
+    // allow injectable override without flagging as unused
+    for (let injectable of this.state.unused.keys())
+      if (injectable[KEY].from === replacedDep[KEY].from)
+        this.state.unused.delete(injectable);
```

Comment:
🤔 this might need to generate a warning as can lead to a situation of "why my mock is not used", as it's not used because it's overridden.

I can imagine situation when you want to override some default deps you might not even be aware of, but in all other cases this really can be configured as an error.

Let's put this conversation aside and later measure how many situations like this we have and why.

Speaking of which - why you added this code? I don't see a test working with this moment, code coverage would really help here.

Reply:
> I can imagine situation when you want to override some default deps you might not even be aware of

Exactly this. A common scenario in our codebase is setting up an array of default injectables, then in a specific test they override it by providing a new injectable of the same dependency by spreading it: `[...defaultDeps, injectable(...)]`. 
While testing this version out, I've found quite a few cases of such behaviour. I think there are two ways of handling this:
1.  the proposed one here, where we skip caring for the overridden one. The side effects are minimal, as the original must be used anyway, so from a dead code perspective there is only potential for an unused mock.
2. complain about this and force devs to use other means, like using `jest.fn()` to override the implementation in the test. This might sound cleaner but it gets tricky in a multi `DiProvider` scenario, where three composition allows closest provider to override dependencies, but return the right mock with `jest.fn()` becomes brittle.

Given the API declares that the overriding strategy is valid, I thought `stats` should be lenient on it too. Also wanna avoid devs abusing `{ track: false }` as that is way worse.

> I don't see a test working with this moment, code coverage would really help here.

[This one is the test to support the use cases](https://github.com/albertogasparin/react-magnetic-di/pull/73/files#diff-496a48b9075007a3923616d8302af7e7e8582b917288a10adedf292775af7127)

Reply:
Thinking - what if allow override only for `{ track: false }` injectables, for others throw/print an error.
May be introduce another flag
- to allow override of given injectable ("it is ok to override me")
- to force override given injectable ("i know what I am doing"), including throwing an error if override did not occur. I would prefer this one.

Reply:
> what if allow override only for { track: false } injectables

I think this will create more problems, as people will use the flag when implementing FF conditions (happens quite often that overrides are because of new FF code path) and then forget to remove the flag on FF cleanup. So `track: false` will spread and unused tracking will become less valuable.

> to force override given injectable

I assume you mean something like: `injectable(bla, blaMock, { override: true })`. 
I get this would be the most correct implementation, however what are the benefits? From an "unused" tracking perspective, if the injectable with override flag set is not used, we will already complain. And complaining about `override` not overriding is semantically correct but a bit too pedantic. At the end of the day it makes no harm.

> Let's put this conversation aside and later measure how many situations like this we have and why.

Yep, let's publish this version and see how many use cases we have